techSpec.js

```diff
@@ -0,0 +1,145 @@
+        Remove list of cats and breed objects from catsRepo.js & breedsRepo.js respectively.   
+        Review if the repositories still need to be classes in this case? 
+        Note that 'id' key in tables is to have PRIMARY KEY constraint (needs to be unique)  
+         
+        2. 
+        Install sqlite3 node module into the root directory of cat api. Import/require sqlite3 into repository files for use.
+
+        We will create a DAO (Data Access Object) which contains a class AppDAO that...
+            1. starts up database connection upon class instantiation:  
+                class AppDAO {
+                    constructor(dbFilePath) {
+                    this.db = new sqlite3.Database(dbFilePath)  
+                    }
+                }
+            2. sets down methods to interface with the database: db.get(), db.each(), db.run() etc
+        This will be the interface between the Repository and the SQLite Database.
+        The DAO class will be instantiated within the main app.js file, when the server is started up via 'npm start'.
+            const dao = new AppDAO('./database.sqlite3')
+        
+            Repository class constructors will take AppDAO class instance upon instantiation:
+                
+            const catRepo = new CatRepository(dao)
+        
+        Repository class methods need to be rewritten to include function calls of the AppDAO class to access and query the database connection.
+        
+        +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+        Each HTTP request (GET, POST, PUT, DEL) requires a different database query command. 
+        How should we deal with this?
+        -----------------------------
+
+        Each request has its own 'database connector' function. This is simplest as request types pass different parameters as arguments.
+        Any parameter passed as argument for use within a SQL command should use parameterization, to prevent against SQL injection. 
+        For eg the HTTP methods GET, DEL & PUT by Id will pass an 'id' argument. For parameterised queries, this should be passed using placeholder: (?)  
+        https://www.w3schools.com/sql/sql_injection.asp
+
+        Here is an example using sqlite's prepared statements: 
+        Eg: https://www.veracode.com/blog/secure-development/how-prevent-sql-injection-nodejs
+        https://stackoverflow.com/questions/40418693/sqlite-how-to-escape-values-to-prevent-sql-injection
+
+        Note also that certain parameters have already been through checks in Router files (isIdvalid() and checkObject())
+
+        Later can be reviewed and DRY-ed from then on. 
+        
+        +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+        All 'database connector' functions begin by opening connection to db, and end with closing the connection: 
```

Comment:
This won't be the case I think - the connection should always remain open, correct?

Reply:
Correct.